include/z64audio.h

```diff
@@ -3,42 +3,68 @@
 
 #define MK_CMD(b0,b1,b2,b3) ((((b0) & 0xFF) << 0x18) | (((b1) & 0xFF) << 0x10) | (((b2) & 0xFF) << 0x8) | (((b3) & 0xFF) << 0))
 
-/**
- * Structs in this repository have primarily been imported from the SM64 Decompilation.
- * Some struct members may be wrong; the symbol '?' next to an offset means the member is a guess.
- */
-
-#define NO_LAYER ((SequenceChannelLayer*)(-1))
-#define NO_CHANNEL ((SequenceChannel*)(-1))
+#define NO_LAYER ((SequenceLayer*)(-1))
 
 #define TATUMS_PER_BEAT 48
 
 #define IS_SEQUENCE_CHANNEL_VALID(ptr) ((u32)(ptr) != (u32)&gAudioContext.sequenceChannelNone)
 
-#define ADSR_STATE_DISABLED 0
-#define ADSR_STATE_INITIAL 1
-#define ADSR_STATE_START_LOOP 2
-#define ADSR_STATE_LOOP 3
-#define ADSR_STATE_FADE 4
-#define ADSR_STATE_HANG 5
-#define ADSR_STATE_DECAY 6
-#define ADSR_STATE_RELEASE 7
-#define ADSR_STATE_SUSTAIN 8
+#define MAX_CHANNELS_PER_BANK 3
 
 #define ADSR_DISABLE 0
 #define ADSR_HANG -1
 #define ADSR_GOTO -2
 #define ADSR_RESTART -3
 
-#define AIBUF_LEN 0xB00
+#define AIBUF_LEN 0x580
+
+typedef enum {
+    /* 0 */ ADSR_STATE_DISABLED,
+    /* 1 */ ADSR_STATE_INITIAL,
+    /* 2 */ ADSR_STATE_START_LOOP,
+    /* 3 */ ADSR_STATE_LOOP,
+    /* 4 */ ADSR_STATE_FADE,
+    /* 5 */ ADSR_STATE_HANG,
+    /* 6 */ ADSR_STATE_DECAY,
+    /* 7 */ ADSR_STATE_RELEASE,
+    /* 8 */ ADSR_STATE_SUSTAIN
+} AdsrStatus;
+
+typedef enum {
+    /* 0 */ MEDIUM_RAM,
+    /* 1 */ MEDIUM_UNK,
+    /* 2 */ MEDIUM_CART,
+    /* 3 */ MEDIUM_DISK_DRIVE
+} SampleMedium;
```

Comment:
```suggestion
    /* 0 */ MEDIUM_RAM,
    /* 1 */ MEDIUM_UNK1,
    /* 2 */ MEDIUM_CART,
    /* 3 */ MEDIUM_DISK_DRIVE
} SampleMedium;
```
This suggestion is optional, but the motivation for it is that MM introduces a new medium with index 5, which is also unknown at this period of time, so `MEDIUM_UNK1` and `MEDIUM_UNK5` are being used, so this would keep things consistent between the two games